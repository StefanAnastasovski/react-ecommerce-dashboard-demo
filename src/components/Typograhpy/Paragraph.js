import React from "react";

import { Typography } from "@mui/material";

const Paragraph = ({ children, ...otherProps }) => {
  return (
    <Typography component="p" {...otherProps}>
      {children}
    </Typography>
  );
};

export default Paragraph;
