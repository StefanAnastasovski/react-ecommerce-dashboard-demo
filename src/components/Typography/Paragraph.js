import React from "react";

import { Typography } from "@mui/material";

const Paragraph = ({ children, ...otherProps }) => {
  return (
    <Typography paragraph mb={0} {...otherProps}>
      {children}
    </Typography>
  );
};

export default Paragraph;
