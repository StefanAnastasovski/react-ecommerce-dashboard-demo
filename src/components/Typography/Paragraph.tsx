import React from "react";

import { Typography } from "@mui/material";
import ParagraphType from "../../@types/ParagraphType";

const Paragraph = ({ children, styleProps, otherProps }: ParagraphType) => {
  return (
    <Typography paragraph mb={0} sx={{ ...styleProps }} {...otherProps}>
      {children}
    </Typography>
  );
};

export default Paragraph;
