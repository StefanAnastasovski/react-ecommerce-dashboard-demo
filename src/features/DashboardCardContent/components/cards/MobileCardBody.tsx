import React from "react";

import ArrowRight from "@mui/icons-material/ArrowForwardOutlined";

import {
  CARD_TITLE_FONT_SIZE_2,
  FONT_WEIGHT_500,
  LINE_HEIGHT_MOBILE_CARD,
  LINK_N_TEXT_FONT_SIZE,
  WHITE_COLOR,
} from "../../../../data/constants";
import DEVICE_IMG from "../../../../assets/images/device.png";

import Paragraph from "../../../../components/Typography/Paragraph";
import GridItem from "../../../../components/Grid/GridItem";
import Image from "../../../../components/Image";
import ExternalLink from "../../../../components/ExternalLink";
import GridContainer from "../../../../components/Grid/GridContainer";
import MobileCardBodyType from "../../../../@types/MobileCardBodyType";

const MobileCardBody = ({
  body: { title, linkContent, href, hasIcon },
  footer: footerIcons,
}: MobileCardBodyType) => {
  const linkIcon = <ArrowRight sx={{ marginLeft: 2.5, fill: WHITE_COLOR }} />;

  const styledParagraph = {
    fontSize: CARD_TITLE_FONT_SIZE_2,
    fontWeight: FONT_WEIGHT_500,
    color: WHITE_COLOR,
    lineHeight: LINE_HEIGHT_MOBILE_CARD,
  };
  const styledExternalLink = {
    color: WHITE_COLOR,
    textDecorationColor: WHITE_COLOR,
  };
  const styledTopLink = {
    marginTop: 2,
    fontSize: LINK_N_TEXT_FONT_SIZE,
  };

  const footerContent =
    footerIcons &&
    footerIcons.map((icon: any) => {
      const { id, href, source, alt } = icon;
      return (
        <GridItem key={id} styleProps={{ alignSelf: "end" }}>
          <ExternalLink
            href={href}
            target="_blank"
            styleProps={styledExternalLink}
          >
            <Image source={source} alt={alt} />
          </ExternalLink>
        </GridItem>
      );
    });

  return (
    <GridContainer spacing="3.65" styleProps={{height: "100%"}}>
      <GridItem styleProps={{ display: "flex", gap: 3 }}>
        <GridItem>
          <Paragraph styleProps={styledParagraph}>{title}</Paragraph>
          <ExternalLink
            href={href}
            target="_blank"
            text={linkContent}
            icon={hasIcon && linkIcon}
            styleProps={{ ...styledExternalLink, ...styledTopLink }}
          />
        </GridItem>
        <GridItem>
          <Image width="105px" source={DEVICE_IMG} alt="device image" />
        </GridItem>
      </GridItem>

      <GridItem
        styleProps={{
          display: "flex",
          flexGrow: 1,
          justifyContent: "space-between",
        }}
      >
        {footerContent}
      </GridItem>
    </GridContainer>
  );
};

export default MobileCardBody;
