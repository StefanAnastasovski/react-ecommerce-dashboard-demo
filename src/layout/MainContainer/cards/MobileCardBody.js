import React from "react";

import ArrowRight from "@mui/icons-material/ArrowForwardOutlined";

import {
  CARD_TITLE_FONT_SIZE_2,
  FONT_WEIGHT_500,
  LINK_N_TEXT_FONT_SIZE,
  WHITE_COLOR,
} from "../../../data/constants";
import DEVICE_IMG from "../../../assets/images/device.png";

import Paragraph from "../../../components/Typography/Paragraph";
import GridContainer from "../../../components/Grid/GridContainer";
import GridItem from "../../../components/Grid/GridItem";
import Image from "../../../components/Image";
import ExternalLink from "../../../components/ExternalLink";

const MobileCardBody = ({
  body: { title, linkContent, href, hasIcon },
  footer: footerIcons,
}) => {
  const linkIcon = <ArrowRight sx={{ marginLeft: 2.5, fill: WHITE_COLOR }} />;

  const styledParagraph = {
    sx: {
      fontSize: CARD_TITLE_FONT_SIZE_2,
      fontWeight: FONT_WEIGHT_500,
      color: WHITE_COLOR,
    },
  };
  const styledExternalLink = {
    sx: {
      display: "flex",
      color: WHITE_COLOR,
      textDecorationColor: WHITE_COLOR,
    },
  };
  const styledTopLink = {
    marginTop: 2,
    fontSize: LINK_N_TEXT_FONT_SIZE,
  };

  const footerContent =
    footerIcons &&
    footerIcons.map((icon) => {
      const { id, href, source, alt } = icon;
      return (
        <GridItem key={id}>
          <ExternalLink href={href} target="_blank" {...styledExternalLink}>
            <Image source={source} alt={alt} />
          </ExternalLink>
        </GridItem>
      );
    });

  return (
    <GridContainer spacing={3.65}>
      <GridItem sx={{ display: "flex", gap: 3 }}>
        <GridItem>
          <Paragraph {...styledParagraph}>{title}</Paragraph>
          <ExternalLink
            href={href}
            target="_blank"
            text={linkContent}
            icon={hasIcon && linkIcon}
            {...{ ...styledExternalLink, ...styledTopLink }}
          />
        </GridItem>
        <GridItem>
          <Image source={DEVICE_IMG} alt="device image" />
        </GridItem>
      </GridItem>

      <GridItem
        sx={{ display: "flex", flexGrow: 1, justifyContent: "space-between" }}
      >
        {footerContent}
      </GridItem>
    </GridContainer>
  );
};

export default MobileCardBody;
