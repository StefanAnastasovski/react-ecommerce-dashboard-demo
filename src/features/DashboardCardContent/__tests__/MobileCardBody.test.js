import React from "react";
import { shallow } from "enzyme";

import APPSTORE_IMG from "../../../assets/images/app-store.png";
import GOOGLEPLAY_IMG from "../../../assets/images/google-play-store.png";
import MobileCardBody from "../components/cards/MobileCardBody";
import Paragraph from "../../../components/Typography/Paragraph";

const props = {
  body: {
    title: "Sell your products on your exclusive APP published on the stores",
    linkContent: "Show More",
    href: "https://www.apple.com/app-store/",
    hasIcon: true,
  },
  footer: [
    {
      id: "stimg1",
      source: APPSTORE_IMG,
      alt: "App Store",
      href: "https://apple.com",
    },
    {
      id: "stimg2",
      source: GOOGLEPLAY_IMG,
      alt: "Google Play Store",
      href: "https://google.com",
    },
  ],
};

describe("SomeComponent component", () => {
  it.skip("Shallow rendering - MatchSnapshot", () => {
    const component = shallow(<MobileCardBody {...props} />);
    expect(component).toMatchSnapshot();
  });

  it.skip("Shallow rendering - match content", () => {
    const component = shallow(<MobileCardBody />);
    expect(component.find(<Paragraph children={props.body.title} />));
  });

  // it("Configure your shop card render testing", () => {
  //   render(<MobileCardBody />);
  //   const element = screen.getByText("Show More");
  //   expect(element).toBeInTheDocument();
  // });
});
