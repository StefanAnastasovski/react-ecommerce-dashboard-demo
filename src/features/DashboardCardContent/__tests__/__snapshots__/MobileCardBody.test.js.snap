// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SomeComponent component Shallow rendering - MatchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MobileCardBody
    body={
      Object {
        "hasIcon": true,
        "href": "https://www.apple.com/app-store/",
        "linkContent": "Show More",
        "title": "Sell your products on your exclusive APP published on the stores",
      }
    }
    footer={
      Array [
        Object {
          "alt": "App Store",
          "href": "https://apple.com",
          "id": "stimg1",
          "source": "app-store.png",
        },
        Object {
          "alt": "Google Play Store",
          "href": "https://google.com",
          "id": "stimg2",
          "source": "google-play-store.png",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <GridItem
          styleProps={
            Object {
              "display": "flex",
              "gap": 3,
            }
          }
        >
          <GridItem>
            <Paragraph
              styleProps={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": "22px",
                  "fontWeight": "500",
                  "lineHeight": "27px",
                }
              }
            >
              Sell your products on your exclusive APP published on the stores
            </Paragraph>
            <ExternalLink
              href="https://www.apple.com/app-store/"
              icon={
                <UNDEFINED
                  sx={
                    Object {
                      "fill": "#FFFFFF",
                      "marginLeft": 2.5,
                    }
                  }
                />
              }
              styleProps={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": "17px",
                  "marginTop": 2,
                  "textDecorationColor": "#FFFFFF",
                }
              }
              target="_blank"
              text="Show More"
            />
          </GridItem>
          <GridItem>
            <Image
              alt="device image"
              source="device.png"
              width="105px"
            />
          </GridItem>
        </GridItem>,
        <GridItem
          styleProps={
            Object {
              "display": "flex",
              "flexGrow": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <GridItem
            styleProps={
              Object {
                "alignSelf": "end",
              }
            }
          >
            <ExternalLink
              href="https://apple.com"
              styleProps={
                Object {
                  "color": "#FFFFFF",
                  "textDecorationColor": "#FFFFFF",
                }
              }
              target="_blank"
            >
              <Image
                alt="App Store"
                source="app-store.png"
              />
            </ExternalLink>
          </GridItem>
          <GridItem
            styleProps={
              Object {
                "alignSelf": "end",
              }
            }
          >
            <ExternalLink
              href="https://google.com"
              styleProps={
                Object {
                  "color": "#FFFFFF",
                  "textDecorationColor": "#FFFFFF",
                }
              }
              target="_blank"
            >
              <Image
                alt="Google Play Store"
                source="google-play-store.png"
              />
            </ExternalLink>
          </GridItem>
        </GridItem>,
      ],
      "spacing": "3.65",
      "styleProps": Object {
        "height": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GridItem>
              <Paragraph
                styleProps={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": "22px",
                    "fontWeight": "500",
                    "lineHeight": "27px",
                  }
                }
              >
                Sell your products on your exclusive APP published on the stores
              </Paragraph>
              <ExternalLink
                href="https://www.apple.com/app-store/"
                icon={
                  <UNDEFINED
                    sx={
                      Object {
                        "fill": "#FFFFFF",
                        "marginLeft": 2.5,
                      }
                    }
                  />
                }
                styleProps={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": "17px",
                    "marginTop": 2,
                    "textDecorationColor": "#FFFFFF",
                  }
                }
                target="_blank"
                text="Show More"
              />
            </GridItem>,
            <GridItem>
              <Image
                alt="device image"
                source="device.png"
                width="105px"
              />
            </GridItem>,
          ],
          "styleProps": Object {
            "display": "flex",
            "gap": 3,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Paragraph
                  styleProps={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": "22px",
                      "fontWeight": "500",
                      "lineHeight": "27px",
                    }
                  }
                >
                  Sell your products on your exclusive APP published on the stores
                </Paragraph>,
                <ExternalLink
                  href="https://www.apple.com/app-store/"
                  icon={
                    <UNDEFINED
                      sx={
                        Object {
                          "fill": "#FFFFFF",
                          "marginLeft": 2.5,
                        }
                      }
                    />
                  }
                  styleProps={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": "17px",
                      "marginTop": 2,
                      "textDecorationColor": "#FFFFFF",
                    }
                  }
                  target="_blank"
                  text="Show More"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Sell your products on your exclusive APP published on the stores",
                  "styleProps": Object {
                    "color": "#FFFFFF",
                    "fontSize": "22px",
                    "fontWeight": "500",
                    "lineHeight": "27px",
                  },
                },
                "ref": null,
                "rendered": "Sell your products on your exclusive APP published on the stores",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "href": "https://www.apple.com/app-store/",
                  "icon": <UNDEFINED
                    sx={
                      Object {
                        "fill": "#FFFFFF",
                        "marginLeft": 2.5,
                      }
                    }
                  />,
                  "styleProps": Object {
                    "color": "#FFFFFF",
                    "fontSize": "17px",
                    "marginTop": 2,
                    "textDecorationColor": "#FFFFFF",
                  },
                  "target": "_blank",
                  "text": "Show More",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Image
                alt="device image"
                source="device.png"
                width="105px"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "alt": "device image",
                "source": "device.png",
                "width": "105px",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GridItem
              styleProps={
                Object {
                  "alignSelf": "end",
                }
              }
            >
              <ExternalLink
                href="https://apple.com"
                styleProps={
                  Object {
                    "color": "#FFFFFF",
                    "textDecorationColor": "#FFFFFF",
                  }
                }
                target="_blank"
              >
                <Image
                  alt="App Store"
                  source="app-store.png"
                />
              </ExternalLink>
            </GridItem>,
            <GridItem
              styleProps={
                Object {
                  "alignSelf": "end",
                }
              }
            >
              <ExternalLink
                href="https://google.com"
                styleProps={
                  Object {
                    "color": "#FFFFFF",
                    "textDecorationColor": "#FFFFFF",
                  }
                }
                target="_blank"
              >
                <Image
                  alt="Google Play Store"
                  source="google-play-store.png"
                />
              </ExternalLink>
            </GridItem>,
          ],
          "styleProps": Object {
            "display": "flex",
            "flexGrow": 1,
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "stimg1",
            "nodeType": "function",
            "props": Object {
              "children": <ExternalLink
                href="https://apple.com"
                styleProps={
                  Object {
                    "color": "#FFFFFF",
                    "textDecorationColor": "#FFFFFF",
                  }
                }
                target="_blank"
              >
                <Image
                  alt="App Store"
                  source="app-store.png"
                />
              </ExternalLink>,
              "styleProps": Object {
                "alignSelf": "end",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Image
                  alt="App Store"
                  source="app-store.png"
                />,
                "href": "https://apple.com",
                "styleProps": Object {
                  "color": "#FFFFFF",
                  "textDecorationColor": "#FFFFFF",
                },
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "alt": "App Store",
                  "source": "app-store.png",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "stimg2",
            "nodeType": "function",
            "props": Object {
              "children": <ExternalLink
                href="https://google.com"
                styleProps={
                  Object {
                    "color": "#FFFFFF",
                    "textDecorationColor": "#FFFFFF",
                  }
                }
                target="_blank"
              >
                <Image
                  alt="Google Play Store"
                  source="google-play-store.png"
                />
              </ExternalLink>,
              "styleProps": Object {
                "alignSelf": "end",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Image
                  alt="Google Play Store"
                  source="google-play-store.png"
                />,
                "href": "https://google.com",
                "styleProps": Object {
                  "color": "#FFFFFF",
                  "textDecorationColor": "#FFFFFF",
                },
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "alt": "Google Play Store",
                  "source": "google-play-store.png",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GridItem
            styleProps={
              Object {
                "display": "flex",
                "gap": 3,
              }
            }
          >
            <GridItem>
              <Paragraph
                styleProps={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": "22px",
                    "fontWeight": "500",
                    "lineHeight": "27px",
                  }
                }
              >
                Sell your products on your exclusive APP published on the stores
              </Paragraph>
              <ExternalLink
                href="https://www.apple.com/app-store/"
                icon={
                  <UNDEFINED
                    sx={
                      Object {
                        "fill": "#FFFFFF",
                        "marginLeft": 2.5,
                      }
                    }
                  />
                }
                styleProps={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": "17px",
                    "marginTop": 2,
                    "textDecorationColor": "#FFFFFF",
                  }
                }
                target="_blank"
                text="Show More"
              />
            </GridItem>
            <GridItem>
              <Image
                alt="device image"
                source="device.png"
                width="105px"
              />
            </GridItem>
          </GridItem>,
          <GridItem
            styleProps={
              Object {
                "display": "flex",
                "flexGrow": 1,
                "justifyContent": "space-between",
              }
            }
          >
            <GridItem
              styleProps={
                Object {
                  "alignSelf": "end",
                }
              }
            >
              <ExternalLink
                href="https://apple.com"
                styleProps={
                  Object {
                    "color": "#FFFFFF",
                    "textDecorationColor": "#FFFFFF",
                  }
                }
                target="_blank"
              >
                <Image
                  alt="App Store"
                  source="app-store.png"
                />
              </ExternalLink>
            </GridItem>
            <GridItem
              styleProps={
                Object {
                  "alignSelf": "end",
                }
              }
            >
              <ExternalLink
                href="https://google.com"
                styleProps={
                  Object {
                    "color": "#FFFFFF",
                    "textDecorationColor": "#FFFFFF",
                  }
                }
                target="_blank"
              >
                <Image
                  alt="Google Play Store"
                  source="google-play-store.png"
                />
              </ExternalLink>
            </GridItem>
          </GridItem>,
        ],
        "spacing": "3.65",
        "styleProps": Object {
          "height": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GridItem>
                <Paragraph
                  styleProps={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": "22px",
                      "fontWeight": "500",
                      "lineHeight": "27px",
                    }
                  }
                >
                  Sell your products on your exclusive APP published on the stores
                </Paragraph>
                <ExternalLink
                  href="https://www.apple.com/app-store/"
                  icon={
                    <UNDEFINED
                      sx={
                        Object {
                          "fill": "#FFFFFF",
                          "marginLeft": 2.5,
                        }
                      }
                    />
                  }
                  styleProps={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": "17px",
                      "marginTop": 2,
                      "textDecorationColor": "#FFFFFF",
                    }
                  }
                  target="_blank"
                  text="Show More"
                />
              </GridItem>,
              <GridItem>
                <Image
                  alt="device image"
                  source="device.png"
                  width="105px"
                />
              </GridItem>,
            ],
            "styleProps": Object {
              "display": "flex",
              "gap": 3,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Paragraph
                    styleProps={
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": "22px",
                        "fontWeight": "500",
                        "lineHeight": "27px",
                      }
                    }
                  >
                    Sell your products on your exclusive APP published on the stores
                  </Paragraph>,
                  <ExternalLink
                    href="https://www.apple.com/app-store/"
                    icon={
                      <UNDEFINED
                        sx={
                          Object {
                            "fill": "#FFFFFF",
                            "marginLeft": 2.5,
                          }
                        }
                      />
                    }
                    styleProps={
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": "17px",
                        "marginTop": 2,
                        "textDecorationColor": "#FFFFFF",
                      }
                    }
                    target="_blank"
                    text="Show More"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Sell your products on your exclusive APP published on the stores",
                    "styleProps": Object {
                      "color": "#FFFFFF",
                      "fontSize": "22px",
                      "fontWeight": "500",
                      "lineHeight": "27px",
                    },
                  },
                  "ref": null,
                  "rendered": "Sell your products on your exclusive APP published on the stores",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "href": "https://www.apple.com/app-store/",
                    "icon": <UNDEFINED
                      sx={
                        Object {
                          "fill": "#FFFFFF",
                          "marginLeft": 2.5,
                        }
                      }
                    />,
                    "styleProps": Object {
                      "color": "#FFFFFF",
                      "fontSize": "17px",
                      "marginTop": 2,
                      "textDecorationColor": "#FFFFFF",
                    },
                    "target": "_blank",
                    "text": "Show More",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Image
                  alt="device image"
                  source="device.png"
                  width="105px"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "alt": "device image",
                  "source": "device.png",
                  "width": "105px",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GridItem
                styleProps={
                  Object {
                    "alignSelf": "end",
                  }
                }
              >
                <ExternalLink
                  href="https://apple.com"
                  styleProps={
                    Object {
                      "color": "#FFFFFF",
                      "textDecorationColor": "#FFFFFF",
                    }
                  }
                  target="_blank"
                >
                  <Image
                    alt="App Store"
                    source="app-store.png"
                  />
                </ExternalLink>
              </GridItem>,
              <GridItem
                styleProps={
                  Object {
                    "alignSelf": "end",
                  }
                }
              >
                <ExternalLink
                  href="https://google.com"
                  styleProps={
                    Object {
                      "color": "#FFFFFF",
                      "textDecorationColor": "#FFFFFF",
                    }
                  }
                  target="_blank"
                >
                  <Image
                    alt="Google Play Store"
                    source="google-play-store.png"
                  />
                </ExternalLink>
              </GridItem>,
            ],
            "styleProps": Object {
              "display": "flex",
              "flexGrow": 1,
              "justifyContent": "space-between",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "stimg1",
              "nodeType": "function",
              "props": Object {
                "children": <ExternalLink
                  href="https://apple.com"
                  styleProps={
                    Object {
                      "color": "#FFFFFF",
                      "textDecorationColor": "#FFFFFF",
                    }
                  }
                  target="_blank"
                >
                  <Image
                    alt="App Store"
                    source="app-store.png"
                  />
                </ExternalLink>,
                "styleProps": Object {
                  "alignSelf": "end",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Image
                    alt="App Store"
                    source="app-store.png"
                  />,
                  "href": "https://apple.com",
                  "styleProps": Object {
                    "color": "#FFFFFF",
                    "textDecorationColor": "#FFFFFF",
                  },
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "alt": "App Store",
                    "source": "app-store.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "stimg2",
              "nodeType": "function",
              "props": Object {
                "children": <ExternalLink
                  href="https://google.com"
                  styleProps={
                    Object {
                      "color": "#FFFFFF",
                      "textDecorationColor": "#FFFFFF",
                    }
                  }
                  target="_blank"
                >
                  <Image
                    alt="Google Play Store"
                    source="google-play-store.png"
                  />
                </ExternalLink>,
                "styleProps": Object {
                  "alignSelf": "end",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Image
                    alt="Google Play Store"
                    source="google-play-store.png"
                  />,
                  "href": "https://google.com",
                  "styleProps": Object {
                    "color": "#FFFFFF",
                    "textDecorationColor": "#FFFFFF",
                  },
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "alt": "Google Play Store",
                    "source": "google-play-store.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
